### PR TITLE
add babel. Fixes #69.

### LINKED ISSUE #69: Adding babel 7
With babel we can write more modern js using feature like esm.
Also jest can use babel too.

### PR DESCRIPTION
**Pull request by gengjiawen**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 node_modules
+.DS_Store
+.idea/
```
Comment (tabrindle) [inline .gitignore]: this should probably be in your own global gitignore

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "envinfo": "dist/cli.js"
   },
   "engines": {
-    "node": ">=4"
+    "node": ">=6"
```
Comment (tabrindle) [inline package.json]: let's experiment with how far back we can support without adding ridiculous numbers of polyfills.

I'll make a support table trying different targets and see what the bundle size is. 

I'd like to *try* to support node 4. It is after EOL but it was an LTS and still used a lot.

File: package.json [hunk 3]
```diff
@@ -51,8 +52,13 @@
   },
   "dependencies": {},
   "devDependencies": {
```
Comment (tabrindle) [inline package.json]: were going to want to remove object.entries and object.values since they should by added via babel

File: .travis.yml [hunk 4]
```diff
@@ -2,6 +2,5 @@ language: node_js
 node_js:
   - "6"
   - "8"
-  - "9"
```
Comment (tabrindle) [inline .travis.yml]: I'm ok with this - 9.x was never LTS and was EOL 2018-06-30

File: .gitignore (hunk 1 above)
Comment (gengjiawen) [reply .gitignore]: .DS_Store is very common.
Also is webstorm.

File: package.json (hunk 2 above)
Comment (gengjiawen) [reply package.json]: Sure, but currently ci has no test for node 4.

File: package.json (hunk 3 above)
Comment (gengjiawen) [reply package.json]: Not quite sure which part you are talking.

File: .travis.yml (hunk 4 above)
Comment (gengjiawen) [reply .travis.yml]: Okay.

File: package.json (hunk 3 above)
Comment (gengjiawen) [reply package.json]: I see.

File: package.json [hunk 5]
```diff
@@ -47,12 +48,19 @@
     ]
   },
   "jest": {
+    "setupFiles": ["./setup.js"],
```
Comment (tabrindle) [inline package.json]: 🤔we shouldn't need this for just babel polyfill - thats what https://www.npmjs.com/package/babel-jest is for.

File: src/envinfo.js [hunk 6]
```diff
@@ -1,16 +1,13 @@
 'use strict';
 
+require('@babel/polyfill');
```
Comment (tabrindle) [inline src/envinfo.js]: no need to add this if we use this config right https://babeljs.io/docs/en/babel-preset-env#usebuiltins

Comment (gengjiawen) [reply src/envinfo.js]: I have tried it, but it not working on node 6.

**Comment by gengjiawen** [comment]
@tabrindle Can you review this again ?

**Comment by gengjiawen** [comment]
@tabrindle ping.

**Comment by tabrindle** [comment]
did a little testing and got the package size down by changing babel configs. I think this is ready to go.

**Comment by gengjiawen** [comment]
@tabrindle squash and merge this ?

**Pull request merged**